src/thor/route_action.cc

```diff
@@ -127,6 +127,27 @@ void remove_edges(const GraphId& edge_id, valhalla::Location& loc, GraphReader&
 namespace valhalla {
 namespace thor {
 
+std::string thor_worker_t::expansion(Api& request) {
+  // tell the path algorithms to track the expansion here
+  expansion_.reset(new rapidjson::Document);
+  // every algorithm gets a default one that is empty
+  expansion_->SetObject();
+  rapidjson::Pointer("/type").Set(*expansion_, "FeatureCollection");
+  rapidjson::Pointer("/properties/algorithm").Set(*expansion_, "none");
+  rapidjson::Pointer("/features/0/type").Set(*expansion_, "Feature");
+  rapidjson::Pointer("/features/0/geometry/type").Set(*expansion_, "MultiLineString");
+  rapidjson::Pointer("/features/0/geometry/coordinates").Create(*expansion_).SetArray();
+  rapidjson::Pointer("/features/0/properties/edge_ids").Create(*expansion_).SetArray();
+  rapidjson::Pointer("/features/0/properties/statuses").Create(*expansion_).SetArray();
```

Comment:
here we fill out a template of the feature collection and the one feature that holds all the linestrings. if its a multileg route we have a problem where the last leg will win on setting which algorithm wsa used. seems like in the future we should ahve a separate feature for each leg